src/js/view/controls/templates/shortcuts-tooltip.js

```diff
@@ -1,16 +1,26 @@
-export default () => (
-    `<div class="jw-shortcuts-tooltip jw-modal jw-reset" title="Keyboard Shortcuts">` +
-        `<span class="jw-hidden" id="jw-shortcuts-tooltip-explanation">` +
-            `Press shift question mark to access a list of keyboard shortcuts` +
-        `</span>` +
-        `<div class="jw-reset jw-shortcuts-container">` +
-            `<div class="jw-reset jw-shortcuts-title">Keyboard Shortcuts</div>` +
-            `<div class="jw-reset jw-shortcuts-tooltip-list">` +
-                `<ul class="jw-shortcuts-tooltip-descriptions jw-reset jw-shortcuts-description">` +
-                `</ul>` +
-                `<ul class="jw-shortcuts-tooltip-keys jw-reset jw-shortcuts-description">` +
-                `</ul>` +
+export default (shortcuts = {}) => {
+    //  Iterate all shortcuts to create list of them. 
+    const shortcutsKey = (key) => `<li><span class="jw-hotkey">${key}</span></li>`;
+    const shortcutsDescription = (description) => `<li><span class="jw-hotkey-description">${description}</span></li>`;
+    const keyList = shortcuts.map(shortcut => shortcutsKey(shortcut.key));
```

Comment:
we can generate both of these in a single loop using a `shortcuts.forEach` instead of iterating through the same list twice

Reply:
i would also suggest we scope the generator functions inside that `forEach` if we go that route since they wouldn't be used anywhere else